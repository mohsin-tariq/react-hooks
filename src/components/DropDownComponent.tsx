import { memo } from 'react'
import { Zones } from '../features/timezone/types'
import { selectors } from '../features/timezone'

interface Props {
  zoneList: Zones[]
  setZoneDetail: React.Dispatch<React.SetStateAction<Zones>>
}

const DropDownComponent = ({ zoneList, setZoneDetail }: Props): JSX.Element => {
  const handleOnChange = (e) => {
    setZoneDetail(
      zoneList.filter(
        (item) => selectors.getSelector(item) === e.target.value
      )[0]
    )
  }
  return (
    <select onChange={(e) => handleOnChange(e)}>
      {zoneList.map((item) => (
        <option key={item.zoneName}>{selectors.getSelector(item)}</option>
      ))}
    </select>
  )
}

export default memo(DropDownComponent)
