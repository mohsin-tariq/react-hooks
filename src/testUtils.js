
/** eslint-disable **/
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'

const initialState = {
  zone: {
    data: [
      {
        countryCode: 'AD',
        countryName: 'Andorra',
        zoneName: 'Europe/Andorra',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
      {
        countryCode: 'AL',
        countryName: 'Albania',
        zoneName: 'Europe/Albania',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
    ],
  },
}

const mockStore = configureMockStore()

function render(
  ui,
  { store = mockStore(initialState), ...renderOptions } = {}
) {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
