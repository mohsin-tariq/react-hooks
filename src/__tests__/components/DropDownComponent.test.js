import React from 'react';
import { render, fireEvent, screen } from '../../testUtils'
import DropDownComponent from '../../components/DropDownComponent'


describe('DropDownComponent', () => {
  it('render Snapshot', () => {
    const { asFragment } = render(<DropDownComponent zoneList= {[
      {
        countryCode: 'AD',
        countryName: 'Andorra',
        zoneName: 'Europe/Andorra',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
      {
        countryCode: 'AL',
        countryName: 'Albania',
        zoneName: 'Europe/Albania',
        gmtOffset: 3600,
        timestamp: 1614084530,
      }
    ]}
    setZoneDetail={jest.fn()}
    />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('verify if element exists', () => {
    render(<DropDownComponent  zoneList= {[
        {
          countryCode: 'AD',
          countryName: 'Andorra',
          zoneName: 'Europe/Andorra',
          gmtOffset: 3600,
          timestamp: 1614084530,
        },
        {
          countryCode: 'AL',
          countryName: 'Albania',
          zoneName: 'Europe/Albania',
          gmtOffset: 3600,
          timestamp: 1614084530,
        }
      ]}
      setZoneDetail={jest.fn()}
     />)
    expect(screen.getByText('Andorra, Europe/Andorra')).toBeInTheDocument()
  })

  it('onchange set value', () => {
    render(<DropDownComponent  zoneList= {[
        {
          countryCode: 'AD',
          countryName: 'Andorra',
          zoneName: 'Europe/Andorra',
          gmtOffset: 3600,
          timestamp: 1614084530,
        },
        {
          countryCode: 'AL',
          countryName: 'Albania',
          zoneName: 'Europe/Albania',
          gmtOffset: 3600,
          timestamp: 1614084530,
        }
      ]}
      setZoneDetail={jest.fn()}
     />)
     fireEvent.change(screen.getByDisplayValue('Andorra, Europe/Andorra'), {
      target: { value: "Albania, Europe/Albania" },
    });
    expect(screen.getByDisplayValue('Albania, Europe/Albania')).toBeInTheDocument()
  })
})

