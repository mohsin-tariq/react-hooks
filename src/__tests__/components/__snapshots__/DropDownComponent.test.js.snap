// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownComponent render Snapshot 1`] = `
<DocumentFragment>
  <select>
    <option>
      Andorra, Europe/Andorra
    </option>
    <option>
      Albania, Europe/Albania
    </option>
  </select>
</DocumentFragment>
`;
